Simple first stab at building core on Windows & Linux

batmat opened the pull request:
Note: stages shouldn't be left in parallel, but not sure what to do if not just remove every stages or so...
Also tried to minimize diff at first.

Also will push some Windows related fixed probably as soon (or maybe before) this is merged. Because this should fail right now for some tests. 

Ultimate goal: make the Java 8 baseline plan actionable.

@slide @daniel-beck @rtyler @jtnord @olivergondza @oleg-nenashev 
@jenkinsci/code-reviewers

oleg-nenashev commented on Jenkinsfile:
```diff
@@ -45,14 +52,24 @@ node('java') {
+        build.call()
+    }
+},       windows: {
+    node('java && windows') {
```
I suppose it will mess up JUnit reports. Maybe the plugin should prefix the reports if possible (or ideally inject OS as a test parameter)

daniel-beck replied:
I don't think this will work if https://github.com/jenkins-infra/documentation/blob/master/ci.adoc#node-labels is still correct.

batmat replied:
Fixed.

batmat replied:
From my testing, it worked, but I suspect you're right, and it's working for me because only Windows has failure in my env. I'm gonna check that.

slide replied:
The other thing is that artifact fingerprinting will fingerprint files from both as the same file name. I have a PR I need to submit that handles this.

slide replied:
Junit reports seemed to work just fine for me.

batmat committed:
First stab at building core on Windows & Linux

Note: stages shouldn't be left in parallel, but not sure what to do if
not just remove every stages or so...

Also tried to minimize diff at first.

slide commented on Jenkinsfile:
```diff
@@ -45,14 +52,24 @@ node('java') {
 }
 
+parallel linux: {
+    node('linux') {
```
This should be 

```
node('java')
```

@rtyler said that Linux nodes are marked as java

batmat replied:
@slide I think this is actually clearer to use `linux` (I initially used `java`) since the [documentation](https://github.com/jenkins-infra/documentation/blob/master/ci.adoc) says indeed 

> `linux` : A Linux (Ubuntu 14.04 LTS) instance (alias of `java`)

batmat commented:
Closing in favor on #2698 which is pretty similar, but handles the artifact archiving better addressing @oleg-nenashev 's concern about files clash and issue with JUnit plugin.

Please review this one from @slide
